534 ez Chapter 4 Veteran Marriages and Veteran Children

vsaleem opened the pull request:
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [ ] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [ ] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)


https://github.com/user-attachments/assets/41c98d21-7cec-4af6-99dc-f585b12bfbef



## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

vsaleem committed:
Add Veteran's previous marriages and children pages

Introduces new configuration files for collecting information about the Veteran's previous marriages and children in the survivors benefits application. Updates form.js to include these new pages in the household information chapter.

vsaleem committed:
Update labels.jsx

vsaleem committed:
cleanup

dgading commented on src/applications/survivors-benefits/config/chapters/04-household-information/veteranMarriagesPages.js:
```diff
@@ -0,0 +1,402 @@
+    ...arrayBuilderItemSubsequentPageTitleUI(
+      "Veteran's previous spouse's name",
+    ),
+    previousSpouseFirst: textUI('First name'),
```
This should be a fullName field, will give the suffix as a select.

dgading commented on src/applications/survivors-benefits/config/chapters/04-household-information/veteranMarriagesPages.js:
```diff
@@ -0,0 +1,402 @@
+      title: 'How did the marriage end?',
+      labels: { DEATH: 'Death', DIVORCE: 'Divorce', OTHER: 'Other' },
+    }),
+    marriageEndedOther: textareaUI({
```
This should be a textUI not textAreaUI

dgading commented on src/applications/survivors-benefits/config/chapters/04-household-information/veteranMarriagesPages.js:
```diff
@@ -0,0 +1,402 @@
+    ...arrayBuilderItemSubsequentPageTitleUI(
+      'When and where did their marriage end?',
+    ),
+    dateOfTermination: currentOrPastDateUI('Date marriage ended'),
```
needs to be memorableDate

dgading commented on src/applications/survivors-benefits/config/chapters/04-household-information/veteranMarriagesPages.js:
```diff
@@ -0,0 +1,402 @@
+    !item?.marriageDate,
+  text: {
+    getItemName: item =>
+      (item && item.previousSpouseLast && `${item.previousSpouseLast}`) ||
```
This needs to be the full name, not just last name

dgading commented on src/applications/survivors-benefits/config/chapters/04-household-information/veteranMarriagesPages.js:
```diff
@@ -0,0 +1,402 @@
+  );
+}
+/** @type {ArrayBuilderOptions} */
+const options = {
```
Need to add the custom max alert items text `alertMaxItems` from figma.

vsaleem commented on src/applications/survivors-benefits/config/chapters/04-household-information/veteranMarriagesPages.js:
```diff
@@ -0,0 +1,402 @@
+    ),
+    marriageDate: currentOrPastDateUI('Date of marriage'),
+    marriagePlace: {
+      ...addressUI({
```
Thank you! I originally went about a round-about way and really over-engineered this >_<;

vsaleem committed:
Refactor previous marriages and children forms

Moved the max items alert for previous marriages to a shared component. Updated previous marriages form to use full name fields and improved location handling for marriages inside and outside the U.S. Switched children questions to use yes/no patterns. Cleaned up and simplified UI and schema logic for better maintainability and user experience.

dgading reviewed (approved):
Looks good code wise.
